File: src/main/java/com/ryderbelserion/discordchat/DiscordChat.java
package com.ryderbelserion.discordchat;

import com.ryderbelserion.discordchat.platform.CommandManager;
import com.ryderbelserion.discordchat.platform.ConfigManager;
import com.ryderbelserion.discordchat.platform.discord.DiscordBot;
import com.ryderbelserion.discordchat.platform.impl.cache.CacheManager;
import com.ryderbelserion.discordchat.platform.impl.storage.Storage;
import com.ryderbelserion.discordchat.platform.impl.storage.StorageFactory;
import org.bukkit.plugin.java.JavaPlugin;

public class DiscordChat extends JavaPlugin {

    private CacheManager cacheManager;
    private Storage storage;
    private DiscordBot bot;

    @Override
    public void onEnable() {
        ConfigManager.load();

        this.cacheManager = new CacheManager();

        this.storage = new StorageFactory().getInstance();

        CommandManager.load();

        this.bot = new DiscordBot();
        this.bot.start();
    }

    @Override
    public void onDisable() {
        this.storage.stop();

        this.bot.stop();
    }

    public CacheManager getCacheManager() {
        return this.cacheManager;
    }

    public Storage getStorage() {
        return this.storage;
    }

    public DiscordBot getBot() {
        return this.bot;
    }
}

File: src/main/kotlin/com/ryderbelserion/discordchat/DiscordChat.kt
package com.ryderbelserion.discordchat

import com.ryderbelserion.discordchat.discord.DiscordBot
import com.ryderbelserion.discordchat.platform.CommandManager
import com.ryderbelserion.discordchat.platform.ConfigManager
import com.ryderbelserion.discordchat.platform.impl.Config
import net.dv8tion.jda.api.entities.Guild
import org.bukkit.plugin.java.JavaPlugin

class DiscordChat : JavaPlugin() {

    private lateinit var bot: DiscordBot
    private lateinit var guild: Guild

    override fun onEnable() {
        ConfigManager.load()

        CommandManager.load()

        val token: String = ConfigManager.config().getProperty(Config.bot_token)

        if (token.isBlank()) {
            this.logger.severe("Token cannot be empty or blank.")

            return
        }

        this.bot = DiscordBot(this)
        this.bot.init()

        //this.guild = this.bot.get()?.getGuildById(ConfigManager.config().getProperty(Config.guild_id))!!
    }

    override fun onDisable() {
        this.bot.onStop()
        this.bot.disable()
    }

    //fun getGuild(): Guild = this.guild
}